### PR TITLE
Fix bug 1659491: Trigger string list search with Enter

### PR DESCRIPTION
**Pull request by mathjazz**
Deployed to stage for user feedback:
https://mozilla-pontoon-staging.herokuapp.com/projects/firefox/all-resources/

### CODE REVIEW

File: frontend/src/modules/search/components/SearchBox.js [hunk 1]
```diff
@@ -330,12 +327,15 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
...
         this.setState({
             search: event.currentTarget.value,
-        }, () => this.updateSearchParams());
+        });
     }
 
-    updateSearchParams = debounce(() => {
-        this.update();
-    }, 500)
+    performSearch = (event: SyntheticKeyboardEvent<HTMLInputElement>) => {
```
Comment (adngdb) [inline frontend/src/modules/search/components/SearchBox.js:333]: This is a bit confusing as it doesn't, strictly speaking, performs the search. I'd call it "handleKeyDown" and add a comment that says it performs the search on Enter.

File: frontend/src/modules/search/components/SearchBox.js [hunk 2]
```diff
@@ -450,6 +450,7 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
                 onBlur={ this.unsetFocus }
                 onChange={ this.updateSearchInput }
                 onFocus={ this.setFocus }
+                onKeyDown={ this.performSearch }
```
Comment (adngdb) [inline frontend/src/modules/search/components/SearchBox.js:453]: The reason for my previous is that here, on reading this line, I think that you perform the search on each key stroke.

File: frontend/src/modules/search/components/SearchBox.js [hunk 3]
```diff
@@ -330,12 +327,15 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
...
+        });
     }
 
-    updateSearchParams = debounce(() => {
-        this.update();
-    }, 500)
+    performSearch = (event: SyntheticKeyboardEvent<HTMLInputElement>) => {
+        // On Enter
+        if (event.keyCode === 13) {
+            this.update()
```
Comment (adngdb) [inline frontend/src/modules/search/components/SearchBox.js:336]: Missing semicolon. :innocent:

**Review by adngdb — APPROVED** [review]

**Comment by mathjazz** [comment]
We've got positive feedback on [Discourse](https://discourse.mozilla.org/t/proposal-to-change-search-behavior-in-the-string-list/65901) and chat. Let's move on!

**Pull request merged**